compiler/src/iree/compiler/Codegen/Common/GPU/GPUMaterializeEncoding.cpp

```diff
@@ -114,11 +114,16 @@ static tensor::ExpandShapeOp reshapeForInnerDims(RewriterBase &rewriter,
   intermediateShape.push_back(numElem);
 
   Location loc = src.getLoc();
-  Type elemType = srcType.getElementType();
+  
+  if (src.getType() != sourceType) {
+    src = rewriter.create<tensor::CastOp>(loc, sourceType, src);
+  }
+
```

Comment:
I don't get why we need the tensor.cast here, but it is fine. We'll kill this method in https://github.com/iree-org/iree/pull/18294 because we will no longer need the dummy reshapes soon.

Reply:
Yeah it is there just for completeness.